pkg/operator/ceph/cluster/controller.go

```diff
@@ -543,6 +543,29 @@ func (c *ClusterController) onDeviceCMUpdate(oldObj, newObj interface{}) {
 	}
 	logger.Debugf("onDeviceCMUpdate new device cm: %+v", newCm)
 
+	oldDevStr, ok := oldCm.Data[discoverDaemon.LocalDiskCMData]
+	if !ok {
+		logger.Warningf("unexpected configmap data")
+		return
+	}
+
+	newDevStr, ok := newCm.Data[discoverDaemon.LocalDiskCMData]
+	if !ok {
+		logger.Warningf("unexpected configmap data")
+		return
+	}
+
+	devicesEqual, err := discoverDaemon.DeviceListsEqual(oldDevStr, newDevStr)
```

Comment:
The orchestration will continue if any of the many properties on the `LocalDisk` struct change, right? I wonder if we could still hit a little noise from that, but we can tackle that another day. For example, we only need to trigger the orchestration if a new device was added, right? If some other property changed such as whether a device has partitions we likely don't need to configure new OSDs.

Reply:
that's right. figuring out exactly what changes should trigger orchestration is still on going. for instance, we ignore disk uuids in the equivalence checks as that was noisy on some test environments, but in other instances a changed uuid might mean a reformatted disk. same for partitions--there are probably noisy changes and non-noisy changes.

btw, you mentioned adding devices explicitly--is the orchestration process responsible for picking up removed devices or is that handled separately? device detection here in discovery picks up removed devices the same as added ones.

Reply:
Right now we only remove OSDs automatically if the whole node is removed. If a device is removed we might just wait for #2741.